app/commands.py

```diff
@@ -314,36 +315,36 @@ async def maplink(ctx: Context) -> str | None:
 async def recent(ctx: Context) -> str | None:
     """Show information about a player's most recent score."""
     if ctx.args:
-        target = app.state.sessions.players.get(name=" ".join(ctx.args))
+        target = await users_repo.fetch_one(name=ctx.args[1])
         if not target:
             return "Player not found."
+        score = await scores_repo.fetch_recent(user_id=target["id"])
     else:
-        target = ctx.player
-
-    score = target.recent_score
+        score = await scores_repo.fetch_recent(user_id=ctx.player.id)
     if not score:
-        return "No scores found (only saves per play session)."
+        return "No scores found."
 
-    if score.bmap is None:
+    beatmap = await Beatmap.from_md5(score["map_md5"])
+    if beatmap is None:
         return "We don't have a beatmap on file for your recent score."
 
-    l = [f"[{score.mode!r}] {score.bmap.embed}", f"{score.acc:.2f}%"]
+    l = [f"[{score['mode']!r}] {beatmap.embed}", f"{score['acc']:.2f}%"]
 
-    if score.mods:
-        l.insert(1, f"+{score.mods!r}")
+    if score["mods"]:
+        l.insert(1, f"+{score['mods']!r}")
 
     l = [" ".join(l)]
 
-    if score.passed:
-        rank = score.rank if score.status == SubmissionStatus.BEST else "NA"
-        l.append(f"PASS {{{score.pp:.2f}pp #{rank}}}")
+    if score["grade"] != "F":
+        rank = score["grade"] if score["status"] == SubmissionStatus.BEST else "NA"
+        l.append(f"PASS {{{score['pp']:.2f}pp #{rank}}}")
```

Comment:
I think `rank` is meant to be the rank on the map, as opposed to the grade

Might need to check if `rank` is a computed property on Score or something